src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -408,24 +444,28 @@ public virtual IEnumerable<IEdmOperationImport> ResolveOperationImports(IEdmMode
         internal static IEdmOperationParameter ResolveOperationParameterNameCaseInsensitive(IEdmOperation operation, string identifier)
         {
             // first look for a case-sensitive match
-            var list = operation.Parameters.Where(parameter => string.Equals(identifier, parameter.Name, StringComparison.Ordinal)).ToList();
-            if (list.Count == 0)
+            var list = operation.Parameters.Where(parameter => string.Equals(identifier, parameter.Name, StringComparison.Ordinal));
+            if (!list.Any())
             {
                 // if no case sensitive, try case-insensitive
-                list = operation.Parameters.Where(parameter => string.Equals(identifier, parameter.Name, StringComparison.OrdinalIgnoreCase)).ToList();
+                list = operation.Parameters.Where(parameter => string.Equals(identifier, parameter.Name, StringComparison.OrdinalIgnoreCase));
             }

```

Comment:
can we combine them as one?

Reply:
As per previous reply, Ordinal string comparison will be cheaper than Ignore case. 

---
In reply to: [362109307](https://github.com/OData/odata.net/pull/1602#discussion_r362109307) [](ancestors = 362109307)